src/main/java/com/minecolonies/coremod/commands/ActionMenuHolder.java

```diff
@@ -22,12 +21,58 @@ public ActionArgument getActionArgument()
     {
         return actionArgument;
     }
-    public Object getValue()
+    @Override
+    public int hashCode()
     {
-        return value;
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((actionArgument == null) ? 0 : actionArgument.hashCode());
+        result = prime * result + ((treeNode == null) ? 0 : treeNode.hashCode());
+        return result;
     }
-    public void setValue(final Object value)
+    @Override
+    public boolean equals(final Object obj)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 8, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)